### ISSUE TITLE
[Combobox] Programmatically selecting a combobox item focuses the component

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-components#5540]
### Actual Behavior

When programmatically selecting a combobox item, focus is moved to the Combobox. This differs from how other form components work, where the acted upon component does not gain focus.

### Expected Behavior

I wouldn't expect the Combobox to gain focus when programmatically setting the selected value of contained combobox items.

### Reproduction Sample

https://codepen.io/mac_and_cheese/pen/rNvXGgz?editors=1000

### Reproduction Steps

1. Open codepen
2. Click the button to set selected on the combobox item.
3. Notice that the combobox has received focus
4. Repeat for input and slider - those components to not receive focus

### Reproduction Version

95

### Esri team

Calcite (design)

**Comment by driskull** [comment]
I think this is the issue; https://github.com/Esri/calcite-components/blob/1b8624e4d2785523bc5670c968065c91b625a3df/src/components/combobox/combobox.tsx#L1027

**Comment by driskull** [comment]
hey @alisonailea don't close the issue, just label it as installed so that it can get verified.

**Comment by geospatialem** [comment]
@alisonailea Can you speak to the skipped tests in `filter`, `input-number`, and `input`? 

The fix works well with `next.642`, but want to make sure the skipped tests are expected.

**Comment by driskull** [comment]
@alisonailea any update on restoring these tests? Do you know why they were skipped?

**Comment by alisonailea** [comment]
> @alisonailea any update on restoring these tests? Do you know why they were skipped?

yep! sorry for the delay. I skipped these tests because they were blocking me from getting my work in. These tests were failing before I did any work. I  will have a PR up soon with some fixes but for the few I still need to skip, I will be added them to the "skipped tests" issue

**Comment by geospatialem** [comment]
> > @alisonailea any update on restoring these tests? Do you know why they were skipped?
> 
> yep! sorry for the delay. I skipped these tests because they were blocking me from getting my work in. These tests were failing before I did any work. I will have a PR up soon with some fixes but for the few I still need to skip, I will be added them to the "skipped tests" issue

@alisonailea Have the skipped tests been added back in? Can you post the PR, or when its merged in, add the PR to this issue?

**Comment by geospatialem** [comment]
A good majority of the skip tests have been addressed via https://github.com/Esri/calcite-components/pull/5885, but there still seems to be a skipped test in `filter`.

@alisonailea It looks like there's one skipped test remaining in [filter.e2e.ts](https://github.com/Esri/calcite-components/blob/master/src/components/filter/filter.e2e.ts#L176-L196) per the discussion in https://github.com/Esri/calcite-components/pull/5774/files#r1030924370.

Reassigning to dev for final incorporation.

**Comment by driskull** [comment]
@alisonailea can you restore the last skipped test?

**Comment by geospatialem** [comment]
> A good majority of the skip tests have been addressed via #5885, but there still seems to be a skipped test in `filter`.
> 
> @alisonailea It looks like there's one skipped test remaining in [filter.e2e.ts](https://github.com/Esri/calcite-components/blob/master/src/components/filter/filter.e2e.ts#L176-L196) per the discussion in https://github.com/Esri/calcite-components/pull/5774/files#r1030924370.
> 
> Reassigning to dev for final incorporation.

@alisonailea Can the skipped tests be restored? Reassigning to dev for final incorporation.

**Comment by alisonailea** [comment]
> @alisonailea Can the skipped tests be restored? Reassigning to dev for final incorporation.

Not without advice from @jcfranco or what appears to be a refactor. Any programmatic changes to the alpha input from Puppeteer causes the tests to error out.

**Comment by alisonailea** [comment]
@geospatialem I just pulled it down and removed the `.skip` on filter.e2e to see what would happen and it is still failing. I don't know why. I didn't touch this code and this test was failing in main before I did any work. 

```
calcite-filter › filter behavior › updates filtered items after filtering

    expect(received).toHaveLength(expected)

    Expected length: 1
    Received length: 4
    Received array:  [{"description": "developer", "metadata": {"favoriteBand": "MetallicA", "haircolor": "red"}, "name": "Harry", "value": "harry"}, {"description": "developer", "metadata": {"favoriteBand": "Radiohead", "haircolor": "black"}, "name": "Matt", "value": "matt"}, {"description": "developer", "metadata": {"favoriteBand": "The Mars Volta", "haircolor": "black"}, "name": "Franco", "value": "franco"}, {"description": "developer", "metadata": {"favoriteBand": "Hippity Hops", "haircolor": "brown"}, "name": "Jon", "value": "jon"}]

      121 |
      122 |   function assertMatchingItems(filtered: any[], values: string[]): void {
    > 123 |     expect(filtered).toHaveLength(values.length);
          |                      ^
      124 |     values.forEach((value) => expect(filtered.find((element) => element.value === value)).toBeDefined());
      125 |   }
 ```

**Comment by jcfranco** [comment]
>  I didn't touch this code and this test was failing in main before I did any work.

Seems like unstable tests at play. @alisonailea Can you create a [follow-up issue](https://github.com/Esri/calcite-components/issues/new?assignees=&labels=testing%2C0+-+new%2Cp+-+high&template=test.yml) to restore the skipped tests?

**Comment by alisonailea** [comment]
Follow up issue created: https://github.com/Esri/calcite-components/issues/6311

**Comment by geospatialem** [comment]
The original issue has been mitigated and verified. With the above follow-up to the `filter` unstable test, will close out this issue.  👍🏻
